### PR TITLE
Add `editor: fold at level <level>` commands

### LINKED ISSUE #5142: Code Folding Levels
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

Going through large files is really difficult.

### Describe the solution you'd like

When I open a large file in VS Code, i can immediately fold all the code at a particular nesting level(Cmd + k + 4 (or a number) to be able to go through the code. It is very helpful to me in my DX.
And also folding in the UI by the line numbers should be possible.
The current folding capability in Zed is too basic to be useful for me

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by JosephTLyons**
Closes https://github.com/zed-industries/zed/issues/5142

Note that I only moved the cursor to the top of the file so it wouldn't jump - the commands work no matter where you are in the file.

https://github.com/user-attachments/assets/78c74ca6-5c17-477c-b5d1-97c5665e44b0

Also, is VS Code doing this right thing here? or is it busted?

https://github.com/user-attachments/assets/8c503b50-9671-4221-b9f8-1e692fe8cd9a

Release Notes:

- Added the ability to fold all blocks of code at a target "level." macOS: `cmd-k, cmd-<number>`, Linux: `ctrl-k, ctrl-<number>`.

### CODE REVIEW

File: assets/keymaps/default-linux.json [hunk 1]
```diff
@@ -313,6 +313,15 @@
       "ctrl-k ctrl-l": "editor::ToggleFold",
       "ctrl-k ctrl-[": "editor::FoldRecursive",
       "ctrl-k ctrl-]": "editor::UnfoldRecursive",
+      "ctrl-k ctrl-1": "editor::FoldAtLevelOne",
```
[the remark this answers is not in the document]
Comment (osiewicz) [reply assets/keymaps/default-linux.json:316]: See e.g. [outline panel](https://github.com/zed-industries/zed/blob/main/crates/outline_panel/src/outline_panel.rs#L64) for an example of such action.

**Comment by JosephTLyons** [comment]
@osiewicz - I remembered the exact downside now that made me want to go with duplicate actions, its that these dynamic ones don't show up in the command palette, so discoverability kind of stinks.

SCR-20241025-mnnt

File: assets/keymaps/default-macos.json [hunk 2]
```diff
@@ -349,7 +349,15 @@
       "alt-cmd-]": "editor::UnfoldLines",
       "cmd-k cmd-l": "editor::ToggleFold",
       "cmd-k cmd-[": "editor::FoldRecursive",
-      "cmd-k cmd-]": "editor::UnfoldRecursive",
```
Comment (ConradIrwin) [inline assets/keymaps/default-macos.json:352]: @JosephTLyons did you nix this on purpose?

**Comment by Tobbe** [comment]
> @osiewicz - I remembered the exact downside that I ran into before doing something similar that made me want to go with duplicate actions, its that these dynamic ones don't show up in the command palette, so discoverability kind of stinks.
> 
> SCR-20241025-mnnt

Saw that this feature was available. Opened up the command palette to try it out. Didn't find it. Went here to figure out how to actually use it.
 → Would be great if these sort of things could be discoverable in the command palette!

**Pull request merged**